File: BACK-END/src/main/java/org/kuenteco/backend/dto/auth/UserDetailDTO.java
package org.kuenteco.backend.dto.auth;

import jakarta.persistence.Version;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kuenteco.backend.dto.image.ImageDTO;
import org.kuenteco.backend.enums.State;
import org.kuenteco.backend.enums.UserType;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDetailDTO {
    @Version private Integer version;
    private ImageDTO image;
    private String username;
    private String email;
    private UserType type;
    private RoleDetailDTO role;
    private State state;
}

File: FRONT-END/lib/dto/auth/response/user_detail_dto.dart
import '../../image/image_dto.dart';

class UserDetailDTO {
  final int? version;
  final ImageDTO? image;
  final String username;
  final String email;
  final String userType;
  final String subscriptionType;
  final String state;

  UserDetailDTO({
    this.version,
    this.image,
    required this.username,
    required this.email,
    required this.userType,
    required this.subscriptionType,
    required this.state,
  });

  factory UserDetailDTO.fromJson(Map<String, dynamic> json) {
    return UserDetailDTO(
      version: json['version'],
      image: json['image'] != null ? ImageDTO.fromJson(json['image']) : null,
      username: json['username'],
      email: json['email'],
      userType: json['userType'],
      subscriptionType: json['subscriptionType'],
      state: json['state'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'version': version,
      'image': image?.toJson(),
      'username': username,
      'email': email,
      'userType': userType,
      'subscriptionType': subscriptionType,
      'state': state,
    };
  }
}